horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/ActionServiceImpl.java

```diff
@@ -127,6 +127,12 @@ public void onNewTest(Test test) {
         executeActions(ActionEvent.TEST_NEW, test.id, test, true);
     }
 
+    @WithRoles(extras = Roles.HORREUM_SYSTEM)
+    @Transactional
+    public void onDatasetLabelsComputed(Integer testId, Object payload) {
```

Comment:
I am expecting this to be of type `Dataset.EventNew`
```suggestion
    public void onDatasetLabelsComputed(Integer testId, Dataset.EventNew payload) {
```